src/fabric.c

```diff
@@ -648,19 +666,19 @@ int DEFAULT_SYMVER_PRE(fi_getinfo)(uint32_t version, const char *node,
 	}
 
 	if (hints && hints->fabric_attr && hints->fabric_attr->prov_name) {
-		util_name = ofi_util_name(hints->fabric_attr->prov_name,
-					  &util_len);
-		core_name = ofi_core_name(hints->fabric_attr->prov_name,
-					  &core_len);
+		prov_vec = ofi_split_and_alloc(hints->fabric_attr->prov_name, ";");
+		if (!prov_vec)
+			return -FI_ENOMEM;
+
+		for (; prov_vec[count]; count++);
```

Comment:
It would make sense to modify ofi_split_and_alloc() to return the number of non-null entries.